segmentstore/server/host/src/test/java/io/pravega/segmentstore/server/host/SegmentStoreFSIntegrationTest.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server.host;
+
+import io.pravega.common.io.FileHelpers;
+import io.pravega.segmentstore.server.store.ServiceBuilder;
+import io.pravega.segmentstore.server.store.ServiceBuilderConfig;
+import io.pravega.segmentstore.server.store.StreamSegmentStoreTestBase;
+import io.pravega.segmentstore.storage.Storage;
+import io.pravega.segmentstore.storage.StorageFactory;
+import io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperConfig;
+import io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperLogFactory;
+import io.pravega.segmentstore.storage.impl.bookkeeper.BookKeeperServiceRunner;
+import io.pravega.segmentstore.storage.impl.filesystem.FileSystemStorageConfig;
+import io.pravega.segmentstore.storage.impl.filesystem.FileSystemStorageFactory;
+import io.pravega.segmentstore.storage.impl.rocksdb.RocksDBCacheFactory;
+import io.pravega.segmentstore.storage.impl.rocksdb.RocksDBConfig;
+import io.pravega.test.common.TestUtils;
+import lombok.val;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+import org.junit.After;
+import org.junit.Before;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * End-to-end tests for SegmentStore, with integrated Storage and DurableDataLog.
+ */
+public class SegmentStoreFSIntegrationTest extends StreamSegmentStoreTestBase {
+    //region Test Configuration and Setup
+
+    private static final int BOOKIE_COUNT = 3;
+    private File baseDir = null;
+    private BookKeeperServiceRunner bkRunner;
+    private CuratorFramework zkClient;
+
+    /**
+     * Starts BookKeeper.
+     */
+    @Before
+    public void setUp() throws Exception {
+        // BookKeeper
+        // Pick random ports to reduce chances of collisions during concurrent test executions.
+        int zkPort = TestUtils.getAvailableListenPort();
```

Comment:
Please unify this class with `SegmentStoreIntegrationTest`, or at least find some sort of inheritance/composition scheme that will allow you to reuse this code. It bothers me a lot to see the exact same code for BK setup/teardown here and in the other file.